core/src/main/java/org/apache/calcite/avatica/remote/AvaticaCommonsHttpClientImpl.java

```diff
@@ -95,29 +103,15 @@ public AvaticaCommonsHttpClientImpl(URL url) {
   }
 
   private void initializeClient() {
-    SSLConnectionSocketFactory sslFactory = null;
-    if (null != truststore && null != truststorePassword) {
-      try {
-        SSLContext sslcontext = SSLContexts.custom().loadTrustMaterial(
-            truststore, truststorePassword.toCharArray()).build();
-
-        final HostnameVerifier verifier = getHostnameVerifier(hostnameVerification);
-
-        sslFactory = new SSLConnectionSocketFactory(sslcontext, verifier);
-      } catch (Exception e) {
-        throw new RuntimeException(e);
-      }
-    } else {
-      LOG.debug("Not configuring HTTPS because of missing truststore/password");
-    }
+    socketFactoryRegistry = this.configureSocketFactories();
+    configureConnectionPool(socketFactoryRegistry);
+    this.authCache = new BasicAuthCache();
+    // A single thread-safe HttpClient, pooling connections via the ConnectionManager
+    this.client = HttpClients.custom().setConnectionManager(pool).build();
+  }
 
-    RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.create();
-    registryBuilder.register("http", PlainConnectionSocketFactory.getSocketFactory());
-    // Only register the SSL factory when provided
-    if (null != sslFactory) {
-      registryBuilder.register("https", sslFactory);
-    }
-    pool = new PoolingHttpClientConnectionManager(registryBuilder.build());
+  protected void configureConnectionPool(Registry<ConnectionSocketFactory> configureSocketFactory) {
+    pool = new PoolingHttpClientConnectionManager(configureSocketFactory);
```

Comment:
meant to be `connectionSocketFactory`?

Reply:
Oh yes, updated it locally but forgot to push it. Will do.